packages/catalog-client/src/CatalogClient.ts

```diff
@@ -456,6 +460,34 @@ export class CatalogClient implements CatalogApi {
     return response.json() as Promise<AnalyzeLocationResponse>;
   }
 
+  /**
+   * {@inheritdoc CatalogApi.streamEntities}
+   */
+  async *streamEntities<
+    T extends StreamEntitiesRequest,
+    R = T extends { mode: 'array' } & StreamEntitiesRequest ? Entity[] : Entity,
```

Comment:
question: Shall we stick to single mode for now? Do you have a use case in mind for `mode: 'array'`?

Reply:
Good question, @freben you think there should be support for getting the entities as batches?

Reply:
I think the numbers kind of speak for having only single-item mode and living with that. It's not so much a use case, more a performance question. And the difference seemed to be pretty small.

I mean, one could argue the other way around that forcing people to write a double `for` loop really is a pretty minimal cost to pay for end users and sidesteps the question while enabling maximum performance potential. But I like the simpler item interface, even if people end up very often doing `Array.from()` on it.

🤔 Hm. This is all a matter of taste at this point.

I vote: Let's go for just single items, and then in the future we can still make this mode addition if we feel that it's necessary. Which maybe it won't be.

Reply:
Backing up a bit, sorry for the noise.

Seems that there are several hundred milliseconds of overhead for large iterations if you wanted an array shaped output in the end. Maybe this actually does make a lot of sense with the mode parameter after all.

Could we make it be `mode?: 'single' | 'pages' | undefined` instead? Only because `'array'` sounds like all of them all at once, not several pages. Or `'batches'` - just that the naming should match the `batchSize`/`pageSize` nomenclature consistently

Reply:
Added the `streamEntityPages` method now 👍

Reply:
I could go for that too. Very low convenience cost for the user, just one method to override, and no foot guns. Maybe it's the reasonable choice.

Reply:
Made the change, please check it out